components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationUserResidentResolverServiceImpl.java

```diff
@@ -60,21 +61,39 @@ public Optional<User> resolveUserFromResidentOrganization(String userName, Strin
             throws OrganizationManagementException {
 
         User resolvedUser = null;
+        String domainName = null;
+        String domainAwareUsername = null;
+
         try {
             if (userName == null && userId == null) {
                 throw handleClientException(ERROR_CODE_NO_USERNAME_OR_ID_TO_RESOLVE_USER_FROM_RESIDENT_ORG);
             }
+            /*
+                Separate out domain name and domain aware username.
+             */
+            if (userName != null && userName.split(DOMAIN_SEPARATOR).length > 1) {
+                domainName = userName.split(DOMAIN_SEPARATOR)[0];
+                domainAwareUsername = userName.split(DOMAIN_SEPARATOR)[0];
```

Comment:
Is this correct?

Reply:
Fixed by - https://github.com/wso2-extensions/identity-organization-management-core/commit/e24babcbb2baa7797fee9295d12bb9220d771f47

Removed this logic and get the user store domain via this method `UserCoreUtil.extractDomainFromName(userName)`.